### ISSUE TITLE
DeprecationWarning/SyntaxError in test_import

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#73879]
BPO | [29693](https://bugs.python.org/issue29693)
--- | :---
Nosy | @brettcannon, @ncoghlan, @benjaminp, @ericsnowcurrently, @serhiy-storchaka, @Carreau, @AnishShah
PRs | <li>python/cpython#421</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-03-03.08:13:21.232>
created_at = <Date 2017-03-02.13:31:48.985>
labels = ['3.7', 'easy', 'type-bug', 'tests']
title = 'DeprecationWarning/SyntaxError in test_import'
updated_at = <Date 2017-03-24.23:05:51.945>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2017-03-24.23:05:51.945>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = True
closed_date = <Date 2017-03-03.08:13:21.232>
closer = 'serhiy.storchaka'
components = ['Tests']
creation = <Date 2017-03-02.13:31:48.985>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 29693
keywords = ['easy']
message_count = 6.0
messages = ['288799', '288820', '288864', '288867', '288872', '290330']
nosy_count = 7.0
nosy_names = ['brett.cannon', 'ncoghlan', 'benjamin.peterson', 'eric.snow', 'serhiy.storchaka', 'mbussonn', 'anish.shah']
pr_nums = ['421']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue29693'
versions = ['Python 3.7']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
```sh
$ ./python -Wa -m test.regrtest test_import
Run tests sequentially
0:00:00 [1/1] test_import
/home/serhiy/py/cpython/Lib/test/test_import/__init__.py:88: DeprecationWarning: invalid escape sequence \(
  self.assertRegex(str(cm.exception), "cannot import name 'i_dont_exist' from 'os' \(.*os.py\)")
/home/serhiy/py/cpython/Lib/test/test_import/__init__.py:96: DeprecationWarning: invalid escape sequence \(
  self.assertRegex(str(cm.exception), "cannot import name 'i_dont_exist' from 'select' \(.*\.(so|pyd)\)")
1 test OK.
```


Total duration: 2 sec
Tests result: SUCCESS

```pytb
$ ./python -We -m test.regrtest test_import
Run tests sequentially
0:00:00 [1/1] test_import
test test_import crashed -- Traceback (most recent call last):
  File "/home/serhiy/py/cpython/Lib/test/libregrtest/runtest.py", line 152, in runtest_inner
    the_module = importlib.import_module(abstest)
  File "/home/serhiy/py/cpython/Lib/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 978, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 675, in exec_module
  File "<frozen importlib._bootstrap_external>", line 782, in get_code
  File "<frozen importlib._bootstrap_external>", line 742, in source_to_code
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/home/serhiy/py/cpython/Lib/test/test_import/__init__.py", line 88
    self.assertRegex(str(cm.exception), "cannot import name 'i_dont_exist' from 'os' \(.*os.py\)")
                                       ^
SyntaxError: invalid escape sequence \(
```


test_import failed

1 test failed:
    test_import

Total duration: 244 ms
Tests result: FAILURE

**Comment by brettcannon** [comment]
Looks like the 'r' prefix is missing on the regex string. Marking this as "easy" if someone wants to try and solve this.

And FYI I double-checked that this is only in 3.7.

**Comment by bb0a482e-e8df-45e0-91a9-f858cc30bdfd** [comment]
I will try to work on this! :)

**Comment by bb0a482e-e8df-45e0-91a9-f858cc30bdfd** [comment]
I'm not able to reproduce the SyntaxError on latest commit 902e9c50e31209e796b6bbe26f8d2f57ec12071b on master branch.

**Comment by serhiy-storchaka** [comment]
Thanks Anish. DeprecationWarning is converted to SyntaxError when run Python with -We.

**Comment by serhiy-storchaka** [comment]
New changeset 15aa4c88f6052af6279e6be3fcd2f968f1c53eae by Serhiy Storchaka (Anish Shah) in branch 'master':
bpo-29693: Fix for DeprecationWarning in test_import (#421)
https://github.com/python/cpython/commit/15aa4c88f6052af6279e6be3fcd2f968f1c53eae
